cmd/release/main.go

```diff
@@ -314,25 +318,33 @@ type git struct {
 }
 
 func (g git) tag(version string) error {
-	// if the tag exists then return nil; if it doesn't create it
-	cmd := exec.Command("git", "rev-parse", "--quiet", "--verify", version)
-	_, err := cmd.CombinedOutput()
+	//TODO: this could probably use some work
+	if oldRev, err := g.tagRevision(version); err != nil {
+		g.logger.Info("non critical error, failed to get tag revision: %v", err)
+	} else {
+		g.logger.Infof("You are forcing the update of a tag reference from %v to the current commit", oldRev)
+	}
+	cmd := exec.Command("git", "tag", "-f", "-s", "-m", fmt.Sprintf("A release of %q for version %q", g.repository, version), version)
+	out, err := cmd.CombinedOutput()
 	if err != nil {
-		// TODO: in go 1.12 use ExitError and ExitCode()
-		// assume this means it doesn't exist
-		cmd = exec.Command("git", "tag", "-s", "-m", fmt.Sprintf("A release of %q for version %q", g.repository, version), version)
-		out, err2 := cmd.CombinedOutput()
-		if err2 != nil {
-			g.logger.Info(string(out))
-		}
-		return err2
+		g.logger.Info(string(out))
 	}
-	return nil
+	return err
+}
+
+func (g git) tagRevision(version string) (string, error) {
+	cmd := exec.Command("git", "rev-parse", version)
+	out, err := cmd.Output()
+	if err != nil {
+		// TODO print stdout as well
+		return "", err
+	}
+	return string(out), err
 }
 
 func (g git) pushTag(version string) error {
-	// TODO(chuckha): this shouldn't exit if it fails because the tag already
-	cmd := exec.Command("git", "push", g.remote, version)
+	g.logger.Infof("You are force pushing a tag: %v\n", version)
+	cmd := exec.Command("git", "push", "-f", g.remote, version)
```

Comment:
Seems a bit scary to me to default to a force push of the tag. It might be better to try and push without force and then prompt the user before doing a force push of the tag if the first push fails.